drgn/helpers/linux/mm.py

```diff
@@ -637,12 +637,20 @@ def compound_head(page: Object) -> Object:
     :return: ``struct page *``
     """
     page = page.read_()
-    head = page.compound_head.read_()
+    try:
+        head = page.compound_head.read_()
+    except AttributeError:
+        # RHEL 7
+        head = page.first_page.address_
```

Comment:
Looking at [`compound_head()` in v4.3](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/linux/mm.h?h=v4.3#n429), shouldn't this be:

```suggestion
    try:
        head = page.compound_head.read_()
    except AttributeError:
        # Before Linux kernel commit 1d798ca3f164 ("mm: make compound_head()
        # robust") (in v4.4), the head page is in page->first_page, and there
        # are no fake head pages.
        return page.first_page.read_() if PageTail(page) else page
```

I don't have a kernel that old available, could you please test it out? If it works, feel free to update the PR.

Reply:
This does work, although i had to also modify `PageTail()` helper in a similar way. Forced-pushed now, please check. Thank you.

Reply:
But this does not work on a RHEL 7 s390x vmcore once `slab_object_info()` is called:

```
  File "/home/onatalen/work/src/crush/crush/helpers/common.py", line 173, in demangle
    obj_info = slab_object_info(ctx.prog, addr)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/drgn/helpers/linux/slab.py", line 534, in slab_object_info
    return _get_slab_cache_helper(slab_cache).object_info(page, slab, addr)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/drgn/helpers/linux/slab.py", line 428, in _get_slab_cache_helper
    return type(slab_cache)
           ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/drgn/helpers/linux/slab.py", line 244, in __init__
    self._slab_cache_size = slab_cache.size.value_()
                            ^^^^^^^^^^^^^^^^^^^^^^^^
_drgn.FaultError: could not read memory from kdump: Excluded page: 0x30038
```